draft-ietf-ipsecme-rfc8229bis.xml

```diff
@@ -1067,12 +1067,37 @@
         </list>
         <xref target="RFC5961" /> discusses how TCP injection attacks can be mitigated.
         </t>
-        <t>
-        Note, that data injection attacks are also possible on IP level (e.g. when IP fragmentation is used)
+
+        <t>Note, that data injection attacks are also possible on IP level (e.g. when IP fragmentation is used)
         resulting in DoS attack even if TCP encapsulation is not used. On the other hand, TCP injection attacks are easier to mount 
         than the IP fragmentation injection attacks, because TCP keeps a long receive window open that’s a sitting target for such attacks. 
         </t>
 
+        <t>If an attacker successfully mounts an injection attack on a TCP connection encapsulating IPsec traffic, 
+        and a Length field in the TCP stream is changed as a result of this attack, then the receiver 
+        in most cases will not be able to correctly identify the boundaries of the following packets in the stream,
+        because it will try to parse an arbitrary data (the result of encryption) as ESP (or IKE) header.
+        This will fail and for this reason all the following packets will be dropped. Eventually the situation should self-recover, but it may 
+        take several minutes and may result in IKE SA deletion and re-creation.
+        </t>
+
+        <t>To speed up the recovery implementations are advised to follow recommendations in <xref target="establish" /> and to close 
+        the TCP connection in case incoming packets contain SPIs which don't match to any of the SAs they have.
+        Once the TCP connection is closed it will be re-created by TCP Originator as described in <xref target="establish" />.
+        </t>
+
+        <t>Alternatively, implementations MAY try to re-sync after they received unknown SPI by searching the TCP stream
+        for a 64 bit binary vector consisting of one of SPIs they know (first 32 bits) and a valid Sequence Number for this SPI 
+        (second 32 bits) and try to validate the ICV of this packet candidate taking the preceding 16 bits as the Length field.
```

Comment:
I think this change is OK, since in English an 'a' article means 'any', right? So we say, that any known SPI will work (so receiver will probably try all of them for every packet candidate).